src/test/java/core/basesyntax/strategy/impl/OperationStrategyImplTest.java

```diff
@@ -0,0 +1,71 @@
+package core.basesyntax.strategy.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.strategy.FruitTransaction;
+import core.basesyntax.strategy.OperationProcessor;
+import core.basesyntax.strategy.OperationStrategy;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+
+@DisplayName("OperationStrategyImpl Test")
+class OperationStrategyImplTest {
+
+    private static OperationStrategy strategy;
+
+    @BeforeAll
+    static void beforeAll() {
+        strategy = new OperationStrategyImpl();
+    }
+
+    @DisplayName("Check method get in Operation Strategy with Balance Processor")
+    @Order(1)
+    @Test
+    void get_getBalanceOperationStrategy_ok() {
+
+        Class<? extends OperationProcessor> actualOperationClass =
+                strategy.get(FruitTransaction.Operation.BALANCE).getClass();
+        Class<? extends OperationProcessor> expectedClass = BalanceProcessor.class;
+        assertEquals(expectedClass, actualOperationClass);
+    }
+
+    @DisplayName("Check method get in Operation Strategy with Purchase Processor")
+    @Order(2)
+    @Test
+    void get_getPurchaseOperationStrategy_ok() {
+        Class<? extends OperationProcessor> actualOperationClass =
+                strategy.get(FruitTransaction.Operation.PURCHASE).getClass();
+        Class<? extends OperationProcessor> expectedClass = PurchaseProcessor.class;
+        assertEquals(expectedClass, actualOperationClass);
+    }
+
+    @DisplayName("Check method get in Operation Strategy with Return Processor")
+    @Order(3)
+    @Test
+    void get_getReturnOperationStrategy_ok() {
+        Class<? extends OperationProcessor> actualOperationClass =
+                strategy.get(FruitTransaction.Operation.RETURN).getClass();
+        Class<? extends OperationProcessor> expectedClass = ReturnProcessor.class;
+        assertEquals(expectedClass, actualOperationClass);
+    }
+
+    @DisplayName("Check method get in Operation Strategy with Supply Processor")
+    @Order(4)
+    @Test
+    void get_getSupplyOperationStrategy_ok() {
+        Class<? extends OperationProcessor> actualOperationClass =
+                strategy.get(FruitTransaction.Operation.SUPPLY).getClass();
+        Class<? extends OperationProcessor> expectedClass = SupplyProcessor.class;
+        assertEquals(expectedClass, actualOperationClass);
+    }
```

Comment:
It's a good point in the educational code, but we don't usually use Reflection API in tests for the real production code (if we speak about the business logic).